pilot/pkg/model/push_context.go

```diff
@@ -502,11 +502,21 @@ func (ps *PushContext) DestinationRule(proxy *Proxy, service *Service) *Config {
 		return nil
 	}
 
-	// search through the DestinationRules in proxy's namespace first
-	if ps.namespaceLocalDestRules[proxy.ConfigNamespace] != nil {
-		if host, ok := MostSpecificHostMatch(service.Hostname,
-			ps.namespaceLocalDestRules[proxy.ConfigNamespace].hosts); ok {
-			return ps.namespaceLocalDestRules[proxy.ConfigNamespace].destRule[host].config
+	// HACK: If the proxy config namespace is same as the root config namespace
```

Comment:
I think the way you describe it as a HACK is too harsh. I think it sort of makes sense to say that we need to guard against the case that a proxy is running in the configured root namespace, whatever it happens to be. The fact that it is the configured "root namespace" means it's supposed to be the last place to look, by definition - if we define it that way :)